packages/puppeteer-core/src/common/bidi/Connection.ts

```diff
@@ -202,7 +212,8 @@ export class Connection extends EventEmitter {
         debugError(object);
       } else {
         this.#maybeEmitOnContext(object);
-        this.emit(object.method, object.params);
+        // SAFETY: We know the method and parameter still match here.
+        this.emit(object.method, object.params as Bidi.BrowsingContext.Info);
```

Comment:
Why exactly `Bidi.BrowsingContext.Info` shouldn't it be something more generic? Like `Bidi.ChromiumBidi.Event`?

Reply:
TypeScript has trouble here because method is a union of all the method types, so it doesn't know how to interpret `object.params`

Reply:
I see, let's type it as `any` as this is not the correct type, comment explains why we do that.